cmd/frontend/internal/httpapi/graphql.go

```diff
@@ -29,6 +32,20 @@ func serveGraphQL(schema *graphql.Schema) func(w http.ResponseWriter, r *http.Re
 
 		r = r.WithContext(trace.WithRequestSource(r.Context(), guessSource(r)))
 
+		if r.Header.Get("Content-Encoding") == "gzip" {
+			reader, err := gzip.NewReader(r.Body)
+			if err != nil {
+				return err
+			}
+			defer reader.Close()
+
+			body, err := ioutil.ReadAll(reader)
+			if err != nil {
+				return err
+			}
+			r.Body = ioutil.NopCloser(bytes.NewReader(body))
```

Comment:
Not sure I'm not missing something, but can't you simplify it to this:

```suggestion
			r.Body = reader
```

Reply:
We're already using a third party middleware library for this. Could we just wrap the graphql handler with it instead? https://sourcegraph.com/github.com/sourcegraph/sourcegraph@23091f7/-/blob/cmd/frontend/internal/app/app.go?subtree=true#L66

Reply:
Yeah, see: https://github.com/sourcegraph/sourcegraph/pull/14391#pullrequestreview-501829175

Reply:
Using that [middleware](https://github.com/nytimes/gziphandler) doesn't actually work, because it only works on the response body, and this is the request.

Reply:
(We can still use this for the response, but I will open another PR for that.)